tiledb/sm/compressors/rle_compressor.h

```diff
@@ -73,8 +78,104 @@ class RLE {
 
   /** Returns the compression overhead for the given input. */
   static uint64_t overhead(uint64_t nbytes, uint64_t value_size);
-};
 
+  /**
+   * Compress variable sized strings in contiguous memory to RLE format
+   *
+   * @tparam T Type of integer to store run legths, must fit max num of
+   * repetitions in input
+   * @tparam P Type of integer to store string sizes, must fit max length input
+   * string
+   * @param input Input in form of a memory contiguous sequence of strings
+   * @param output RLE-encoded output as a series of [run
+   * length|string_size|string] items. Memory is allocated and owned by the
+   * caller
+   */
+  template <class T, class P>
+  static void compress(
+      const span<std::string_view> input, span<std::byte> output) {
+    if (input.empty() || output.empty())
+      return;
+
+    const uint64_t run_size = sizeof(T);
+    const uint64_t str_len_size = sizeof(P);
+    const uint64_t max_run_length = std::numeric_limits<T>::max();
+
+    T run_length = 1;
+    auto out_offset = &output[0];
+    auto previous = input[0];
+    for (uint64_t in_index = 1; in_index < input.size(); in_index++) {
```

Comment:
Using `decltype(input.size()) in_index` means that you don't need to worry about what size type `size()` returns.